xclim/subset.py

```diff
@@ -125,11 +182,412 @@ def func_checker(*args, **kwargs):
                     kwargs[lon] -= 360
                 else:
                     kwargs[lon][kwargs[lon] < 0] -= 360
+
         return func(*args, **kwargs)
 
     return func_checker
 
 
+def wrap_lons_and_split_at_greenwich(func):
```

Comment:
Isn't this solving an issue that regionmask has already solved ?

Reply:
Yes, I see that Regionmask this function as well. There were a few reasons why I didn't consider integrating Regionmask. This is worth a longer discussion, but it looks as though the maintainer of Regionmask is considering some of the approaches shown here. Perhaps we can work on that project and integrate it into our work here?

Reply:
Considering that the subsetting functions affect a lot of people, I think it's important we discuss and document high-level decisions before implementation.